### PR TITLE
Upgrade to Alpine 3.11

### PR DESCRIPTION
**Pull request by J0WI**

### CODE REVIEW

**Comment by tianon** [comment]
Looks like we need to replicate https://gitlab.alpinelinux.org/alpine/aports/blob/840f665bfeaa4de03d66d8a6d69ceb3a331e3bf6/main/postgresql/postgresql.pre-install now that our `postgres` user is not part of `alpine-baselayout`. :sweat_smile:
(Similar to https://github.com/docker-library/php/blob/bb80b576dd6da2fd73760f5dd3a3a89980805703/Dockerfile-alpine.template#L25-L32)

Do you think there's a version of Alpine/musl where https://github.com/docker-library/postgres/issues/582 might become relevant for Alpine variants as well?

**Comment by yosifkit** [comment]
Looks like we are getting two versions of `llvm-libs` and so the build is confused. [`clang`](https://pkgs.alpinelinux.org/package/v3.11/main/x86_64/clang) is the problem, since it is newer and depends on `llvm9-libs`. I guess swap everything to `llvm9`?

```console
(40/61) Installing llvm8-libs (8.0.1-r2)
(41/61) Installing llvm8 (8.0.1-r2)
(42/61) Installing llvm8-dev (8.0.1-r2)
(43/61) Installing llvm9-libs (9.0.0-r1)
```

**Comment by J0WI** [comment]
> Do you think there's a version of Alpine/musl where #582 might become relevant for Alpine variants as well?

I'm not aware of any breaking changes in musl-libc, but I haven't tested all cases myself.

**Comment by tianon** [comment]
Ok, cool. :+1:

(I also did a little digging and couldn't find anything that made me believe this was a problem for this bump, but it's still possible in the future.  Hopefully some of the ideas in https://wiki.postgresql.org/wiki/Collations get better traction so this can be a non-issue for future PostgreSQL versions via ICU or the like. :+1:)

File: 10/alpine/Dockerfile [hunk 1]
```diff
@@ -1,17 +1,13 @@
...
-#   /etc/group:34:postgres:x:70:
-# the home directory for the postgres user, however, is not created by default
-# see https://github.com/docker-library/postgres/issues/274
-RUN set -ex; \
-	postgresHome="$(getent passwd postgres)"; \
-	postgresHome="$(echo "$postgresHome" | cut -d: -f6)"; \
-	[ "$postgresHome" = '/var/lib/postgresql' ]; \
-	mkdir -p "$postgresHome"; \
-	chown -R postgres:postgres "$postgresHome"
+# 70 is the standard uid/gid for "postgres" in Alpine
```
Comment (yosifkit) [inline 10/alpine/Dockerfile]: Would it make sense to unify explicit user ids across the Debian and Alpine images, like [redis](https://github.com/docker-library/redis/pull/203/files#diff-d3ac17aae384a7e89cc0b7645bd38cacR4) and in #93? Can the database directory be used by Alpine if it was created by Debian and vise-versa?

Comment (tianon) [reply 10/alpine/Dockerfile]: In theory, but I think locale issues will wreak havoc on anyone who does so.

File: Dockerfile-alpine.template [hunk 2]
```diff
@@ -1,17 +1,13 @@
...
-	postgresHome="$(getent passwd postgres)"; \
-	postgresHome="$(echo "$postgresHome" | cut -d: -f6)"; \
-	[ "$postgresHome" = '/var/lib/postgresql' ]; \
-	mkdir -p "$postgresHome"; \
-	chown -R postgres:postgres "$postgresHome"
+# 70 is the standard uid/gid for "postgres" in Alpine
+# https://git.alpinelinux.org/aports/tree/main/postgresql/postgresql.pre-install?h=3.11-stable
+RUN set -eux; \
+	addgroup -g 70 -S postgres; \
+	adduser -u 70 -S -D -G postgres postgres; \
```
Comment (tianon) [inline Dockerfile-alpine.template]: I think we should also set `/var/lib/postgresql` as the home directory here, like we do in Debian:

https://github.com/docker-library/postgres/blob/0d0485cb02e526f5a240b7740b46c35404aaf13f/Dockerfile-debian.template#L18

(and like Alpine has done previously and still does in https://git.alpinelinux.org/aports/tree/main/postgresql/postgresql.pre-install?h=3.11-stable)

**Comment by smellman** [comment]
Why this PR is not merged?

**Pull request merged**